### ISSUE TITLE
Can't install packages on corporate pc

### ISSUE THREAD

**Issue by gitboy16** [JuliaLang/Pkg.jl#2516]
Hi, we are using R and Python on corporate windows desktop behind a proxy. We wanted to try the Julia language but it has proven difficult to install any package. We are not admin on our computers. Please see error message below.

```
(@1.6) pkg> add DataFrames
      Resolving package versions...
           Cloning [bd369af6-aec1-5ad0-b16a-f7cc5008161c] Tables from https://github.com/JuliaData/Tables.jl.git
ERROR: failed to clone from https://github.com/JuliaData/Tables.jl.git, error: GitError(Code:ERROR, Class:OS, failed to get server certificate: The handle is in the wrong state for the requested operation)
```
Would you be able to help?
Please let me know if you need further informations. Thank you.

**Comment by digital-carver** [comment]
(Not affiliated with Pkg.jl, just someone who came across this, but:)

It looks like the usual cause of this error is Windows attempting to use TLS version 1.1 or lower which Github doesn't support (see: https://stackoverflow.com/questions/48985995/gitkraken-and-github-failed-to-get-server-certificate-the-handle-is-in-the-wr). 

* Are you able to clone https repositories outside of Julia on the system? Does `git clone https://github.com/JuliaData/Tables.jl.git` work on the commandline? 
* What version of Windows is this on? Is the 'Step-by-step fix' mentioned in the StackOverflow answer applicable for it (i.e. is it Windows Server 2012, or Windows Server 2008, or somehow Windows 7)?

**Comment by gitboy16** [comment]
- git clone works outside of Julia. All git commands work on the command line. We use either git portable or the one provided by the IT department and both work fine.
- we are running on windows 10. I can't run the step by step fix of stackoverflow because I am not administrator of my machine. I would need to check with IT.
- We are using git with Python and R and everything is working fine. Might it be a problem of git configuration in Julia? I am not sure which version of git is Julia using. Can we get Julia to use a different version of git?

Thank you

**Comment by fredrikekre** [comment]
You probably need to set `JULIA_NO_VERIFY_HOSTS` and maybe `JULIA_ALWAYS_VERIFY_HOSTS` depending on your setup, see https://github.com/JuliaLang/NetworkOptions.jl#verify_host

**Comment by gitboy16** [comment]
It does not seem to be working. Is there a procedure to git clone packages outside of Julia, save them in the right directory and install them manually?

**Comment by fredrikekre** [comment]
What did you try and what is the error message?

Pkg does not use git to install packages, it uses https downloads.

**Comment by gitboy16** [comment]
I have tried the below (with * and without):
```
export JULIA_NO_VERIFY_HOSTS="**.github.com" export JULIA_SSL_NO_VERIFY_HOSTS="github.com" export JULIA_SSH_NO_VERIFY_HOSTS="github.com"
```

**Comment by fredrikekre** [comment]
Okay, what does
```
using Downloads
Downloads.request("https://github.com/JuliaLang/Example.jl/archive/refs/tags/v0.5.3.tar.gz")
```
error with with those set?

**Comment by barucden** [comment]
Could the proxy be the cause of that? Maybe you (or your IT) have configured git to use the proxy using `git config`, so then clones  via `git` work. Downloads.jl uses libcurl to perform the actual request, so setting environment variable `http_proxy` accordingly might help (see [libcurl docs](https://curl.se/libcurl/c/CURLOPT_PROXY.html))?

**Comment by gitboy16** [comment]
I have tried to set up `http_proxy` and `https_proxy` to the same as `git` but it still fails. The error is however different:
```
ERROR: failed to clone from https://github.com/Julia/DataFrames.jl.git, error: GitError(Code:ERROR, Class:Net, malformed URL 'http://user:pwd@proxy.net:8080/')
```

**Comment by fredrikekre** [comment]
Did you include a trailing slash? See https://github.com/JuliaLang/julia/issues/33111 and https://github.com/JuliaLang/julia/issues/33111#issuecomment-678218751 in particular.

**Comment by gitboy16** [comment]
Yes I did. `/` at the end.

**Comment by barucden** [comment]
Imho, these are two different issues. 

How about forcing Pkg to use the CLI git by setting the variable `JULIA_PKG_USE_CLI_GIT=true`? Although, I am not sure about the consequences.

**Comment by gitboy16** [comment]
I did `ENV["JULIA_PKG_USE_CLI_GIT"]=true` and the error message is still the same.

**Comment by StefanKarpinski** [comment]
Could not resolve host is a new one to me. Seems like DNS resolution isn't working for libcurl some reason?

**Comment by gitboy16** [comment]
Please let me know if you need any information that can help you solve this issue. Currently we can't use Julia at work. We can work with R and install packages and same with Python but would really like to try Julia...thank you

**Comment by StefanKarpinski** [comment]
> I did `ENV["JULIA_PKG_USE_CLI_GIT"]=true` and the error message is still the same.

Unless you are on a development version of Julia, this won't have any effect since this feature was added recently. 



> I have tried to set up `http_proxy` and `https_proxy` to the same as `git` but it still fails. The error is however different:
> 
> ```
> ERROR: failed to clone from https://github.com/Julia/DataFrames.jl.git, error: GitError(Code:ERROR, Class:Net, malformed URL 'http://user:pwd@proxy.net:8080/')
> ```

What are the values of the `http_proxy` and `https_proxy` environment variables? The error message includes the URL `http://user:pwd@proxy.net:8080/` which seems strange.

**Comment by gitboy16** [comment]
Hi, I tried again this morning. I changed my password because I had some character like #$%() in my password so I removed it and kept it simple just letters and number as I noticed that `npm` was struggling with that.
I have done the following:
-Open cmd
-SET http_proxy=http://user:password@proxy.net:8080/ and same for https_proxy
-launched julia and tried to install a package.
-I got the same error that posted at the very beginning.
-I tried to set up JULIA_PKG_USE_GIT to true...the error is still the same
I tried to get the error message with the URL http://user:password@proxy.net:8080/ but it does show anymore. I always get the error message at the very beginning. :-(

**Comment by StefanKarpinski** [comment]
Ok, what I'm trying to get at is this: is your `http_proxy` setting literally `http://user:password@proxy.net:8080/` or is that just a placeholder for what you're actually setting it to that you've also replaced for the real value in the error messages?

It's really difficult to pin down what's going on since the error message keeps changing and I have no idea if that's because of the network or OS are changing or because your Julia client is actually configured differently than it was last time — you didn't see the "malformed URL" error until after setting several proxy environment variables.

**Comment by gitboy16** [comment]
Ah sorry, yes it is a place holder, I hidded my user name and password. The OS is not changing I am running on windows 10. Yes I tried to get the malformed URL error again but I always get the failed to get server certificate one.

**Comment by gitboy16** [comment]
Never giving up...downloaded the nightly version of julia. Then I have set up HTTP(S)_PROXY like I said above, then `JULIA_PKG_USE_CLI_GIT=true`
, and it worked. I could install everything I want. :-) 
When I did pkg> add DataFrames ...the colors were not properly rendering but if i do Pkg.add("DataFrames") ...colors were fine... I guess I will have to wait for the next stable release of julia, in the meantime I will play with the dev version. Thank you all for your help. It has been painful but it worked.

**Comment by StefanKarpinski** [comment]
Great, I'm glad that worked for you. 🎉

**Comment by gitboy16** [comment]
Thank you @StefanKarpinski ,  is there a page showing when the next releases are going to happen. I could see in the changelog of Pkg that JULIA_PKG_USE_CLI_GIT was being mentionned for 1.7?

**Comment by gitboy16** [comment]
Yes that is the file I found but it was not mentionning any date. I guess I will have to just wait. Tks

**Comment by StefanKarpinski** [comment]
We don't announce release dates, we only announce feature freeze dates and then the release takes however long it takes.
